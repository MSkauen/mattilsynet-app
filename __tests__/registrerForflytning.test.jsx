import React from "react";
import ReactDOM from "react-dom";
import { describe, expect, it } from "@jest/globals";
import { act } from "react-dom/test-utils";
import {MemoryRouter} from "react-router";
import GetRegistrerForflytning from "../src/client/RegistrerForflytning";

async function render(component) {
  const container = document.createElement("div");
  document.body.appendChild(container);
  await act(async () => {
    ReactDOM.render(<MemoryRouter>{component}</MemoryRouter>, container);
  });
  return container;
}

const forflytningApi = {
    async getAllRegistrations() {
        return (
            [
                {
                    "produksjonsplassid": 1000000,
                    "kommunenummer": "0001",
                    "gaardsnummer": 1,
                    "bruksnummer": 1,
                    "bygningsnummer": -1,
                    "koordinater": null,
                    "koordinatsystem": null,
                    "opprettetdato": "2022-10-17",
                    "lastchanged": "2022-10-17"
                },
                {
                    "produksjonsplassid": 1000001,
                    "kommunenummer": "0001",
                    "gaardsnummer": 3,
                    "bruksnummer": 1,
                    "bygningsnummer": -1,
                    "koordinater": "{\"type\": \"Point\", \"crs\": {\"type\": \"name\", \"properties\": {\"name\": \"EPSG:4326\"}}, \"coordinates\": [11.348141, 60.845944]}",
                    "koordinatsystem": "EPSG:4326",
                    "opprettetdato": "2022-10-17",
                    "lastchanged": "2022-10-17"
                },
                {
                    "produksjonsplassid": 1000002,
                    "kommunenummer": "0001",
                    "gaardsnummer": 2,
                    "bruksnummer": 1,
                    "bygningsnummer": -1,
                    "koordinater": "{\"type\": \"Point\", \"crs\": {\"type\": \"name\", \"properties\": {\"name\": \"EPSG:4326\"}}, \"coordinates\": : [11.631629, 60.31758]}",
                    "koordinatsystem": "EPSG:4326",
                    "opprettetdato": "2022-10-17",
                    "lastchanged": "2022-10-17"
                }
            ]
        );
    }
};

describe("get all registered forflytninger", () => {
  it("show registered forflytninger on dom", async () => {

    const container = await render(<GetRegistrerForflytning forflytningApi={ forflytningApi } />);
    expect(container.innerHTML).toMatchSnapshot();
    expect(document.querySelector('option[value^="1000000"]').value).toEqual("1000000");
  });
})