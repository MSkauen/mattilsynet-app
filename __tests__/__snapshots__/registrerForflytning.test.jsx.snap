// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get all registered forflytninger show registered forflytninger on dom 1`] = `"<div id="main" class="RegistrerForflytning"><header id="head" class="RegistrerForflytning-header"><img id="logo" src="" class="RegistrerForflytning-logo" alt="logo"><div id="form" class="RegistrerForflytning-form"><form><br><label>Identifiser dyrehold:<select required="" name="dyrehold"><option value="" disabled="" selected="">Velg dyrehold ...</option><option value="2512341">Dyrehold 1</option><option value="2512342">Dyrehold 2</option><option value="2512343">Dyrehold 3</option><option value="2512344">Dyrehold 4</option></select><br></label><label>Individ merke(r):<input required="" type="number" name="fra" style="width: 80px;" value="fra">&nbsp;<input required="" type="number" name="til" style="width: 80px;" value="til"></label><br><label>Produksjonsplass:<select id="sted" required="" name="sted"><option value="" disabled="" selected="">Velg produksjonsplass ...</option><option value="1000000">0001-1/1</option><option value="1000001">0001-3/1</option><option value="1000002">0001-2/1</option></select></label><br><input type="submit" name="registrer"></form></div></header></div>"`;
